### PR TITLE
snARS Integration

### PR DESCRIPTION
**Pull request by jalextowle**
# Description

This PR adds the snARS integration. This yield source is a staking vault for the nARS stable coin. Some interesting things about this yield source are that `asBase = true` is not supported on withdrawal since the user will not receive adequate base (nARS is instead locked in an escrow contract, which we don't want to subject our users to).

### CODE REVIEW

File: deployments.json [hunk 1]
```diff
@@ -1453,41 +1453,6 @@
             "address": "0xFcdaF9A4A731C24ed2E1BFd6FA918d9CF7F50137",
             "timestamp": "2024-09-23T17:24:41.079Z"
         },
-        "ElementDAO ERC4626 Hyperdrive Deployer Coordinator_ERC4626HyperdriveCoreDeployer": {
```
Comment (mcclurejt) [inline deployments.json:1456]: are these going to get replaced?

**Review by mcclurejt — APPROVED** [review]
LGTM

File: tasks/deploy/config/base/snars-182day.ts [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+            "HyperdriveFactory",
+            hre.hyperdriveDeploy.deployments.byName(BASE_FACTORY_NAME).address,
+        );
+        return {
+            baseToken: NARS_ADDRESS_BASE,
+            vaultSharesToken: SNARS_ADDRESS_BASE,
+            circuitBreakerDelta: parseEther("0.075"),
+            minimumShareReserves: parseEther("0.001"),
+            minimumTransactionAmount: parseEther("0.001"),
+            positionDuration: parseDuration(SIX_MONTHS),
```
Comment (jrhea) [inline tasks/deploy/config/base/snars-182day.ts:67]: didn't we talk about this being 1 month or 3 months?

File: hardhat.config.base.ts [hunk 3]
```diff
@@ -40,6 +41,7 @@ const config: HardhatUserConfig = {
                     BASE_MORPHO_BLUE_CBETH_USDC_182DAY,
                     BASE_MOONWELL_ETH_182DAY,
                     BASE_STK_WELL_182DAY,
+                    BASE_SNARS_182DAY,
```
Comment (jrhea) [inline hardhat.config.base.ts:44]: if the term is going to be 1 month or 3 months we should rename this

**Review by jrhea — CHANGES_REQUESTED** [review]

**Review by jrhea — APPROVED** [review]

**Comment by jalextowle** [comment]
Closed because these changes were incorporated into #1189

**Pull request closed without merging**